
##First of all, for using this scripts, you need to be sure of what is a Chip-seq analysis: ChIP-sequencing is
a method used to analyze protein interactions with DNA. ChIP-seq combines chromatin immunoprecipitation (ChIP)
with massively parallel DNA sequencing to identify the binding sites of DNA-associated proteins.
It can be used to map global binding sites precisely for any protein of interest. In this case, scripts were done
to identify the binding site of transcriptions factors in a concrete genome.

##This scripts are a really interesting tool that were done by Julia Pérez and Fernando Rodríguez for that people
that need to perfom a Chip-seq analysis with any kind of organism and number of samles. It may help you to
reach your conclusion in less time because it works in an automated way.

##The first thing you may do it is to make a comprehension reading of this README. Here you will find all the necessary
information to perform the analysis without making mistakes. Read carefully and enjoy!

##You could find instructions in each scripts .sh to understand how the processing has been done, but the only file you will
need to modify is the one called *Params.txt* that is specifict for each analysis. You will find two files of params in the
folder called *Test*. The one called *Paramscp.txt* is used in the case you have the samples and you only need to copy
The other one that is called *Paramsdownload.txt* is used if you need to download the data from the web.

Do not modify the name of the param, only the information that is writtrn after the :.

##In this part, the meaning of each param will be explained in order to you to change with your data:

-
